app/jobs/content_event_job.rb

```diff
@@ -0,0 +1,39 @@
+# A generic job for sending events about a generic files to a user and their followers.
+#
+# @attr [String] generic_file_id  the id of the file the event is specified for
+#
+class ContentEventJob < EventJob
+  attr_accessor :generic_file_id
+
+  def initialize(generic_file_id, depositor_id)
+    super(depositor_id)
+    self.generic_file_id = generic_file_id
+  end
+
+  def run
+    super
+
+    log_generic_file_event
+  end
+
+  def generic_file
+    @generic_file ||= GenericFile.load_instance_from_solr(generic_file_id)
```

Comment:
@mjgiarlo The files them selves are not changing, with the exception of the proxy deposit, so yes it still works, and in the proxy I overrode the generic_fie method to use find.  It is part of my war on using find when it is not needed to just automatically replace it.